File: meatromney/src/main/java/be/ordina/hp/pact/annhannah/application/PizzaController.java
package be.ordina.hp.pact.annhannah.application;

import be.ordina.hp.pact.annhannah.domain.MeatRomneyPizza;
import org.springframework.core.ParameterizedTypeReference;
import org.springframework.hateoas.Resources;
import org.springframework.http.HttpMethod;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.client.RestTemplate;

import java.util.List;
import java.util.stream.Collectors;

@RestController
public class PizzaController {

    private final RestTemplate restTemplate;


    public PizzaController(RestTemplate restTemplate) {
        this.restTemplate = restTemplate;
    }

    @GetMapping("/")
    public List<MeatRomneyPizza> getMeatApprovedPizzas() {
        final ResponseEntity<Resources<MeatRomneyPizza>> responseEntity = restTemplate.exchange(
                "http://localhost:8080/pizzas",
                HttpMethod.GET, null,
                new ParameterizedTypeReference<Resources<MeatRomneyPizza>>() {
                });
        return responseEntity.getBody().getContent().stream().filter(pizza -> pizza.isMeatRomneyApproved()).collect(Collectors.toList());
    }
}

File: meatromney/src/main/java/be/ordina/hp/pact/meatromney/application/PizzaController.kt
package be.ordina.hp.pact.meatromney.application

import be.ordina.hp.pact.meatromney.domain.MeatRomneyPizza
import org.springframework.core.ParameterizedTypeReference
import org.springframework.hateoas.Resources
import org.springframework.http.HttpMethod
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.client.RestTemplate
import java.util.stream.Collectors

@RestController
class PizzaController(private val restTemplate: RestTemplate) {

    val meatApprovedPizzas: List<MeatRomneyPizza>
        @GetMapping("/")
        get() {
            val responseEntity = restTemplate.exchange(
                    "http://localhost:8080/pizzas",
                    HttpMethod.GET, null,
                    object : ParameterizedTypeReference<Resources<MeatRomneyPizza>>() {

                    })
            return responseEntity.body.content.stream().filter { pizza -> pizza.isMeatRomneyApproved }.collect(Collectors.toList())
        }
}
